pkg/cmd/adm/restart.go

```diff
@@ -1,157 +1,213 @@
+
+			if err := restarter(ctx, nonOperatorDeployment, f, ioStreams); err != nil {
+				return err
+			}
+			//check the rollout status
+			ctx.Printlnf("Checking the status of the rolled out deployment %v", nonOperatorDeployment.Name)
+			if err := checker(ctx, f, ioStreams, "toolchain.dev.openshift.com/provider=codeready-toolchain"); err != nil {
+				return err
+			}
+			return nil
+		}
+		ctx.Printlnf("Found only autoscaling-buffer deployment in namespace %s , which is not required to be restarted", ns)
 	}
 
-	return restartDeployment(ctx, hostClient, hostNamespace, deployments.Items[0].Name)
+	return nil
 }
 
-func printExistingDeployments(term ioutils.Terminal, cl runtimeclient.Client, ns string) error {
-	deployments := &appsv1.DeploymentList{}
-	if err := cl.List(context.TODO(), deployments, runtimeclient.InNamespace(ns)); err != nil {
+func deleteAndWaitForPods(ctx *clicontext.CommandContext, cl runtimeclient.Client, deployment appsv1.Deployment) error {
+	ctx.Printlnf("Listing the pods to be deleted")
+	//get pods by label selector from the deployment
+	pods := corev1.PodList{}
+	selector, _ := metav1.LabelSelectorAsSelector(deployment.Spec.Selector)
+	if err := cl.List(ctx, &pods,
+		runtimeclient.MatchingLabelsSelector{Selector: selector},
+		runtimeclient.InNamespace(deployment.Namespace)); err != nil {
 		return err
 	}
-	deploymentList := "\n"
-	for _, deployment := range deployments.Items {
-		deploymentList += fmt.Sprintf("%s\n", deployment.Name)
+	ctx.Printlnf("Starting to delete the pods")
+	//delete pods
+	for _, pod := range pods.Items {
+		pod := pod // TODO We won't need it after upgrading to go 1.22: https://go.dev/blog/loopvar-preview
+		if err := cl.Delete(ctx, &pod); err != nil {
+			return err
+		}
 	}
-	term.PrintContextSeparatorWithBodyf(deploymentList, "Existing deployments in %s namespace", ns)
 	return nil
+
 }
 
-func scaleToZero(cl runtimeclient.Client, namespacedName types.NamespacedName) (int32, error) {
-	// get the deployment
-	deployment := &appsv1.Deployment{}
-	if err := cl.Get(context.TODO(), namespacedName, deployment); err != nil {
-		return 0, err
+func restartNonOperatorDeployments(ctx *clicontext.CommandContext, deployment appsv1.Deployment, f cmdutil.Factory, ioStreams genericclioptions.IOStreams) error {
+
+	o := kubectlrollout.NewRolloutRestartOptions(ioStreams)
+
+	if err := o.Complete(f, nil, []string{"deployment"}); err != nil {
+		panic(err)
 	}
-	// keep original number of replicas so we can bring it back
-	originalReplicas := *deployment.Spec.Replicas
-	zero := int32(0)
-	deployment.Spec.Replicas = &zero
 
-	// update the deployment so it scales to zero
-	return originalReplicas, cl.Update(context.TODO(), deployment)
+	o.Resources = []string{"deployment/" + deployment.Name}
+
+	if err := o.Validate(); err != nil {
+		panic(err)
+	}
+	ctx.Printlnf("Running the rollout restart command for non-Operator deployment %v", deployment.Name)
```

Comment:
btw, in all the messages that are printed out, we should rather talk about OLM-based deployments, not operator deployments as it can be confusing.
You can also rename the function and vars to make it clear in the code as well.

Reply:
I had already done that previously, but i understood this [comment](https://github.com/kubesaw/ksctl/pull/79#discussion_r1781318452) a bit differently and changed all the names , changed it back now